src/api.ts

```diff
@@ -55,59 +55,76 @@ export interface PaginationOptions {
   offset?: number;
 }
 
-function restFetchClient(baseUrl: string, defaultOptions: RequestInit = {}) {
-  async function _fetch<T>(
+class FetchClient {
+  private readonly baseUrl: string;
+  private readonly defaultOptions: RequestInit;
+
+  constructor(baseUrl: string, defaultOptions: RequestInit = {}) {
+    this.baseUrl = baseUrl;
+    this.defaultOptions = defaultOptions;
+  }
+
+  setBearerToken(token: string) {
+    this.defaultOptions.headers = {
+      ...this.defaultOptions.headers,
+      Authorization: `Bearer ${token}`,
+    };
+  }
+
+  protected async fetch<T>(
     endpoint: string,
     options: RequestInit = {},
+    responseType: "json" | "text" = "json",
```

Comment:
I'm wondering whether there's a chance that the value passed in for this would contradict the generic type `T`—and whether there's a way the `response[responseType]` below could rely on the generic type `T` instead. I don't know whether TypeScript supports that. I found [this SO answer](https://stackoverflow.com/a/53321037) in which someone _uses_ the generic type within the function, but they are still using it as a _type_, and not checking its "value." Anyway, this is more something I'm curious about. It's not something I want to prevent a merge over. Maybe we can have a docstring that guides the caller to ensure they are consistent.

Reply:
I believe the answer to your immediate question is: no, you can't use the generic type `T` for that kind of check. It's a TypeScript-only construct that won't be there at runtime. 

But I'm glad you pointed that out because I hadn't meant to commit that stuff. I was chasing down an issue and I _thought_ it had to do with the string response you get from `/api/me`. Turned out to be a red herring, but I forgot to remove this debugging code.